db/data/20220520103603_add_new_allocation_subject_general_sciences.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class AddNewAllocationSubjectGeneralSciences < ActiveRecord::Migration[6.1]
+  def up
+    allocation_subject = AllocationSubject.create(name: AllocationSubjects::GENERAL_SCIENCES)
```

Comment:
```suggestion
allocation_subject = AllocationSubjects::GENERAL_SCIENCES
subject_specialisms = ALLOCATION_SUBJECT_SPECIALISM_MAPPING[allocation_subject]

# Remove old mapping of existing specialisms
SubjectSpecialism.where(name: subject_specialisms).delete_all

# Move subject specialisms under new allocation subject
AllocationSubject.create(name: allocation_subject).tap do |as|
  subject_specialisms.each do |subject_specialism|
    as.subject_specialisms.create(name: subject_specialism)
  end
end
```

Reply:
That makes a lot of sense, will make the change now